Lib/Src/CJsonParser.cpp

```diff
@@ -0,0 +1,339 @@
+    // Note: To learn about range-based loops:
+    // https://www.techiedelight.com/iterate-over-characters-string-cpp/
+    for (char current_char : string)
+    {
+        switch (current_char)
+        {
+            case '0':
+            case '1':
+            case '2':
+            case '3':
+            case '4':
+            case '5':
+            case '6':
+            case '7':
+            case '8':
+            case '9':
+                if (current_token.type == STRING ||
+                    current_token.type == INTEGER ||
+                    current_token.type == FLOAT)
+                {
+                    current_token.text.append(1, current_char);
+                }
+                else
+                {
+                    endToken(current_token);
+                    current_token.text.append(1, current_char);
+                    current_token.type = INTEGER;
+                }
+                break;
+            case '.':
+                if (current_token.type == INTEGER)
+                {
+                    current_token.text.append(1, current_char);
+                    current_token.type = FLOAT;
+                }
+                else
+                {
+                    endToken(current_token);
+                    current_token.text.append(1, current_char);
+                    current_token.type = INVALID;
+                }
+                break;
+            case ' ':
+            case '\t':
+                if (current_token.type == STRING)
+                {
+                    current_token.text.append(1, current_char);
+                }
+                else
+                {
+                    endToken(current_token);
+                }
+                break;
+            case '\n':
+            case '\0':
+            case '\r':
+                endToken(current_token);
+                break;
+            case ':':
+                addSpecialChar(current_token, COLON, current_char);
+                break;
+            case '{':
+                addSpecialChar(current_token, CURLY_OPEN, current_char);
+                break;
+            case '}':
+                addSpecialChar(current_token, CURLY_CLOSE, current_char);
+                break;
+            case '[':
+                addSpecialChar(current_token, ARRAY_OPEN, current_char);
+                break;
+            case ']':
+                addSpecialChar(current_token, ARRAY_CLOSE, current_char);
+                break;
+            case ',':
+                addSpecialChar(current_token, COMMA, current_char);
+                break;
+
+            case '"':
+                if (current_token.type != STRING)
+                {
+                    endToken(current_token);
+                    current_token.type = STRING;
+                }
+                else
+                {
+                    endToken(current_token);
+                }
+                break;
+
+            default:
```

Comment:
Requires explanation when you expect to encounter this branch and what is being done to handle it gracefully.